### PR TITLE
=str #16938 Enable more tests in GraphConcatSpec

### PR DESCRIPTION
**Pull request by bantonsson**
Refs #16938

### CODE REVIEW

**Comment by bantonsson** [comment]
I made this comment in the issue:

> I don't see the issue with the tests, since they test the opposite wiring of each other and should cover both materialization cases, unless our materialization order is random.

**Comment by bantonsson** [comment]
So @drewhk explained how the materializer works and invalidated my previous comment. I quickly double checked this case and it works since the graphs created only contains 3 submodules that are put into a Set that becomes an optimized Set3, that traverses in insertion order.

I added a comment to the tests.

**Comment by patriknw** [comment]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphConcatSpec.scala [hunk 1]
```diff
@@ -103,8 +103,11 @@ class GraphConcatSpec extends TwoStreamsSetup {
       subscriber2.expectSubscriptionAndError(TestException)
     }
 
-    "work with one delayed failed and first nonempty publisher" in {
-      pending // FIXME: This relies on materialization order!!
+    "work with one nonempty and one delayed failed publisher" in {
+      // This test and the next one are materialization order dependent and rely on the fact
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphConcatSpec.scala]: we might break this ordering though, but I guess the comment is enough so the first one encountering the issue can modify it accordingly.

**Comment by drewhk** [comment]
LGTM

**Pull request merged**